### ISSUE TITLE
Midea Dishwasher ST 5.31 wi

### ISSUE THREAD

**Issue by TheCountRobel** [kovapatrik/homebridge-midea-platform#104]
Hi,

Can you add support for the dishwasher? 

| Name  | Type | Model | ID | Ver. | Token / Key | Local Ip | Add / Update |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| midea_e1_0287  | Unknown | 7600000K | 152832116811136 | 3 | XXX | 192.168.0.100 | Not supported |

Thanks, 
Alex

**Comment by kovapatrik** [comment]
Yes, I can totally do that!

**Comment by kovapatrik** [comment]
In the meantime, could you tell me what features you would need? You can also send screenshots with a little description about them.

**Comment by kovapatrik** [comment]
I've released basic support for the dishwasher in `v1.2.0-beta.1`, but only the power on/off has been implemented yet. I would need more information from you about your needs.

**Comment by salalalalaman** [comment]
Thank you @kovapatrik  for working on this! I downloaded your beta plugin and was able to import the dishwasher into Homekit. However, its stating that the device is not supported in Homekit. I am not seeing the on/off button. I can also provide further information in terms of features and testing. Thanks!
![CAFD9786-9582-4C52-9AA1-D4974F6F011E](https://github.com/user-attachments/assets/fd91601d-e29c-4f39-b850-7b6da86b3ccb)

**Comment by kovapatrik** [comment]
I think I missed out some required attributes from the implementation. I will fix it tomorrow, and write a message here to notify you about it!

**Comment by TheCountRobel** [comment]
Hello @kovapatrik , thank you for your work! 
Maybe it would be nice to not only start/stop the dishwasher but also show the remaining time like in the screenshot in #87  
If it is possible to have some kind of sensor that sends a push notification when the dishwasher is finished

best regards from Austria, Alex

**Comment by kovapatrik** [comment]
I've released `v1.2.0-beta.2`, now HomeKit shouldn't identify this as an unsupported device and also I've added the remaining time as well. I can also add a sensor if there is no possibility of creating an automation based on the remaining time. Could you please try this with the new version and report back?

**Comment by salalalalaman** [comment]
Thanks @kovapatrik for the updated plugin! Unfortunately, I'm still running into the same problem where the device is not supported. I am on ios 18.1 beta so i am not sure if that may be a contributing factor

**Comment by kovapatrik** [comment]
Hm weird. The beta version of iOS shouldn't be a factor in my opinion but who knows? Are there any errors or warnings about the dishwasher in Homebridge itself?

**Comment by salalalalaman** [comment]
I checked the homebridge logs and although it had discovered the dishwasher and connected it to homebridge successfully there was one instance that it couldn't find the device 

`[01/09/2024, 16:30:07] [homebridge-midea-platform] [Dishwasher] Device not found (id: 150633093278542), will retry every 60 seconds`

**Comment by kovapatrik** [comment]
Hm, it's not much though. Can you please try to remove this device from the Homebridge cache and try again? If it's not working, I would wait for someone who doesn't use a beta version of iOS, or if you have a MacBook or an iPad you can try this on those as well.

**Comment by TheCountRobel** [comment]
I've got following Message in Homekit, IOS 17.6.1
![WhatsApp Bild 2024-09-01 um 19 14 05_21cfbdc7](https://github.com/user-attachments/assets/8116d298-9e03-4acf-b819-f32da7d11445)

**Comment by kovapatrik** [comment]
I found the missing piece. This should now work with `v1.2.0-beta.4`

**Comment by TheCountRobel** [comment]
yes now its working! Is it possible to set a default washing program?
At the moment it will turn on the Dishwasher but it can't start, Looks like no washing program is selected.
![WhatsApp Bild 2024-09-01 um 19 26 22_8874b4bf](https://github.com/user-attachments/assets/a866a7cc-5d4a-4b1e-925e-98d3dd27077d)
![WhatsApp Bild 2024-09-01 um 19 28 20_c1a05b46](https://github.com/user-attachments/assets/b4605306-952d-4e0b-b3d4-ae0a34d102fb)

**Comment by kovapatrik** [comment]
Yes, I can, I just don't know how to display the options in Apple Home as there are no such things in HomeKit for now like a multi-value switch. I can make a switch for each washing program if that's okay with you.

**Comment by TheCountRobel** [comment]
sure should work for me. If i could choose the default programm in the Homebridge Plugin I would also be happy

**Comment by kovapatrik** [comment]
You mean that there should be a start button or something, which will start the default program selected in the plugin's configuration?

**Comment by mgoeppl** [comment]
Many plugins use TVs as multi-value switches, e.g. homebridge-iRobot. 
I’m personally using the MF200 washer, which can’t even set the program via its own app, and even starting is only possible remotely if it is actively configured each time after the washer is turned on and the program is already set. So I personally don’t think it’s possible tbh?

**Comment by kovapatrik** [comment]
My Midea AC has a few features which are not available directly through the official application but by using the plugin, I can use them too (like turning on/off the LEDs on it). But you might be right on this one, it's worth a try tho.

Thanks the information about the TV! I will check that service out.

**Comment by kovapatrik** [comment]
I'm a little out of the loop about this issue now. 
So if I have it right, you need a dropdown or something in the plugin, which will set the default program of the dishwasher, and upon you start it, that program will be started. Am I right?

Also, I would like to deep dive a bit in the code of the dishwasher. So, if you could, can you please download the latest beta, do a discovery with the default Midea SmartHome profile, and then in the discovered devices table, the model numbers will be highlighted as a button which will download the code for that specific model. Can you please download that file and send me it? You can upload it right here, or send me in email (`kovapatrik@gmail.com`). If you choose the email one, please notify me about it here, it might get in my spams.

**Comment by TheCountRobel** [comment]
Excactly, at the moment i can start the device but i have to start the washing manually on the device, where can i find the downloaded code?

**Comment by kovapatrik** [comment]
I made a documentation about how to do this: https://github.com/kovapatrik/homebridge-midea-platform/blob/main/docs/download_lua.md

**Comment by kovapatrik** [comment]
@TheCountRobel Hello! I'm back fixing all my issues, and I want to ask if you can assist with this one.

**Comment by R-Rennert** [comment]
Hi,

Can you add support for Midea SV 7.03M14B dishwasher?

name: midea_e1_0084
model: 760001CC

I have added to Apple HomeKit, but can only set on and off. I can‘t see status if on, like time, door opened or closed and dishwasher Running.

Thanks! Ronny

[e1_760001CC.lua.txt](https://github.com/user-attachments/files/21748776/e1_760001CC.lua.txt)
